adapters/readpeak/readpeak.go

```diff
@@ -0,0 +1,163 @@
+package readpeak
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Readpeak adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+
+	requestCopy := *request
+	var rpExt openrtb_ext.ImpExtReadpeak
+	var imps []openrtb2.Imp
+	for i := 0; i < len(requestCopy.Imp); i++ {
+		if requestCopy.Imp[i].Native == nil && requestCopy.Imp[i].Banner == nil {
+			continue
+		}
```

Comment:
`static/bidder-info/readpeak.yaml` specifies either `banner` or `Native` is required. 
prebid server core will ensure either `banner` or `native` is present else will return error before adapter code is being invoked

Reply:
Removed the check